ross/shaft_element.py

```diff
@@ -1192,3 +1192,1133 @@ def section(
+        self.o_d = (float(odl) + float(odr)) / 2
+        self.i_d = (float(idl) + float(idr)) / 2
+        self.idl = float(idl)
+        self.odl = float(odl)
+        self.idr = float(idr)
+        self.odr = float(odr)
+        self.color = self.material.color
+
+        self.alpha = float(alpha)
+        self.beta = float(beta)
+
+        # # A_l = cross section area from the left side of the element
+        # # A_r = cross section area from the right side of the element
+        # A_l = np.pi * (odl ** 2 - idl ** 2) / 4
+        # A_r = np.pi * (odr ** 2 - idr ** 2) / 4
+        # self.A_l = A_l
+        # self.A_r = A_r
+
+        # # Second moment of area of the cross section from the left side
+        # # of the element
+        # Ie_l = np.pi * (odl ** 4 - idl ** 4) / 64
+
+        # outer = self.odl ** 2 + self.odl * self.odr + self.odr ** 2
+        # inner = self.idl ** 2 + self.idl * self.idr + self.idr ** 2
+        # self.volume = np.pi * (self.L / 12) * (outer - inner)
+        # self.m = self.material.rho * self.volume
+
+        # Timoshenko kappa factor determination, based on the diameters relation
+        if self.__is_circular():
+            kappa = (6 * (1 + self.material.Poisson) ** 2) / (
+                7 + 12 * self.material.Poisson + 4 * self.material.Poisson ** 2
+            )
+        elif self.__is_thickwall():
+            a = (self.idl + self.idr) / 2
+            b = (self.odl + self.odr) / 2
+            v = self.material.Poisson
+            kappa = (6(*a ** 2 + b ** 2) ** 2 * (1 + v) ** 2) / (
+                7 * a ** 4
+                + 34 * a ** 2 * b ** 2
+                + 7 * b ** 4
+                + v(12 * a ** 4 + 48 * a ** 2 * b ** 2 + 12 * b ** 4)
+                + v ** 2 * (4 * a ** 4 + 16 * a ** 2 * b ** 2 + 4 * b ** 4)
+            )
+        else:
+            kappa = (1 + self.material.Poisson) / (2 + self.material.Poisson)
+
+        self.kappa = kappa
+
+    def __is_circular(self):
+        return self.idl == 0 and self.idr == 0
+
+    def __is_thickwall(self):
+        p = (((self.odl + self.odr) / 2) - ((self.idl + self.idr) / 2)) / (
+            (self.odl + self.odr) / 2
+        )
+        return p >= 0.2
+
+    def __eq__(self, other):
+        """
+        Equality method for comparasions
+
+        Parameters
+        ----------
+        other : obj
+            parameter for comparasion
+
+        Returns
+        -------
+        True if other is equal to the reference parameter.
+        False if not.
+
+        Example
+        -------
+        >>> from ross.materials import steel
+        >>> le = 0.25
+        >>> i_d = 0
+        >>> o_d = 0.05
+        >>> shaft1 = ShaftElement(
+        ...        le, i_d, o_d, steel, rotary_inertia=True, shear_effects=True
+        ... )
+        >>> shaft2 = ShaftElement(
+        ...        le, i_d, o_d, steel, rotary_inertia=True, shear_effects=True
+        ... )
+        >>> shaft1 == shaft2
+        True
```

Comment:
I think this example won't run.
You need to point out the material argument. This way, it's like `idr=steel`
Try this way:
```
>>> shaft1 = ShaftElement(
...        le, i_d, o_d, material=steel, rotary_inertia=True, shear_effects=True
... )
>>> shaft2 = ShaftElement(
...        le, i_d, o_d, material=steel, rotary_inertia=True, shear_effects=True
... )
```